cluster-autoscaler/utils/kubernetes/client.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"net/url"
+
+	"k8s.io/autoscaler/cluster-autoscaler/config"
+
+	kube_client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+const (
+	failedToBuildConfigErr       = "Failed to build config"
+	failedToParseK8sUrlErr       = "Failed to parse Kubernetes url"
+	failedToBuildClientConfigErr = "Failed to build Kubernetes client configuration"
+)
+
+// CreateKubeClient creates kube client based on AutoscalingOptions.KubeClientOptions
+func CreateKubeClient(opts *config.KubeClientOptions) kube_client.Interface {
```

Comment:
nit: Do we need to pass this via pointer? The struct is small and this won't be called frequently, so performance is not an issue and when passed by value it is clear at the call site that the data under pointer is not getting modified. You also can stop taking the address of `KubeClientOps` then.

Reply:
> this won't be called frequently

Agreed. 

>  when passed by value it is clear at the call site that the data under pointer is not getting modified

I see what you mean. 

Honestly, I don't have a strong opinion here. The argument for pointer is, the number of fields in `KubeClientOptions` may increase in the future ([there are many rest config options](https://github.com/kubernetes/autoscaler/blob/57f2814267399e0900a70d567a7813089a3cfa5b/cluster-autoscaler/vendor/k8s.io/client-go/rest/config.go#L50-L146) which aren't supported through `KubeClientOptions` as of now). Changing this to a pointer later would mean a change across core and cloudproviders (i.e., places which use this function) and getting review from cloudprovider maintainers.

Reply:
Let's see if @elmiko also has any comment on this as a cloudprovider maintainer.

Reply:
I will wait until Monday and make the change to pass by value if there are no strong opinions on this.

Reply:
i think what @x13n is proposing sounds reasonable, passing by pointer does make me wonder if some mutation is happening.